flask_rebar/compat.py

```diff
@@ -0,0 +1,60 @@
+import sys
+from distutils.version import StrictVersion
+
+import marshmallow
+
+
+PY2 = int(sys.version_info[0]) == 2
+MARSHMALLOW_VERSION = StrictVersion(marshmallow.__version__).version
+MARSHMALLOW_MAJOR_VERSION = MARSHMALLOW_VERSION[0]
+
+
+if MARSHMALLOW_MAJOR_VERSION == 2:
+    def set_data_key(field, key):
+        field.load_from = key
+        field.dump_to = key
+        return field
+
+
+    def get_data_key(field):
+        if field.dump_to:
+            return field.dump_to
+        elif field.load_from:
+            return field.load_from
+        return field.name
+
+
+    def load(schema, data):
+        schema.strict = True
+        data, _ = schema.load(data)
+        return data
+
+
+    def dump(schema, data):
+        schema.strict = True
+        data, _ = schema.dump(data)
+        return data
+
+    def exclude_unknown_fields(schema):
+        return schema
+
+else:
```

Comment:
This PR is currently getting rid of the runtime compatibility check on the marshmallow version (previously it was logging a warning if found to be outside expectation). 

If code throughout this package is going to branch off `if MALLOW_V2` and then only have an `else` clause that *assumes* V3, is it worth keeping a marshmallow version compatibility check at runtime?

Reply:
Now that we've specified the dependency properly in `setup.py`, tools like `pip` should give the warning for us. In the case where someone install marshmallow>=4 despite this warning, we should give our best effort support.

So I think the `else` is fine for now 🤷‍♂️